src/main/java/subway/line/Line.java

```diff
@@ -45,4 +62,41 @@ public void update(String name, String color) {
         this.name = name;
         this.color = color;
     }
+
+    public void addSection(Section section) {
+        this.sections.add(section);
+        this.downStation = section.getDownStation();
+    }
+
+    public void deleteSection(Long stationId) {
+        validateMultiSection();
+        validateStationOnLine(stationId);
+        validateEndStation(stationId);
+
+        int lastIndex = sections.size() - 1;
+        Section section = this.sections.get(lastIndex);
+        this.downStation = section.getUpStation();
+    }
+
+    private void validateEndStation(Long stationId) {
+        if (!stationId.equals(downStation.getId())) {
+            throw new SubwayException(ErrorCode.NOT_END_STATION);
+        }
+    }
+
+    private void validateStationOnLine(Long stationId) {
+        boolean stationFound = sections.stream().anyMatch(section ->
```

Comment:
```suggestion
        boolean stationFound = sections.hasStation(stationId);
```

이렇게 바꾸면 어떨까요?
section에서 값을 꺼내서 비교할 필요가 없어지지 않을가요?

Reply:
리뷰 감사합니다. 따로 함수로 추출하였습니다.